### ISSUE TITLE
matching filetypes with a dot in the name

### ISSUE THREAD

**Issue by selimober** [sukima/xmledit#48]
I want to use the plugin with javascript files, jsx in fact.
I'm using https://github.com/mxw/vim-jsx plugin which sets the filetype of `*.js` files to:
```
:set ft?
filetype=javascript.jsx
```
I've tried to copy `html.vim` under `~/.vim/plugged/xmledit/ftplugin`  to `javascript.vim`, `javascript.jsx.vim`, `javascriptjsx.vim` and `jsx.vim` to no avail.

Otherwise the plugin works on html and xml without a problem.

**Comment by marlun** [comment]
I was also thinking about wanting this to work. Maybe there's something @mxw could do to help with this too?

**Comment by mxw** [comment]
@selimober—Is the issue that your .vim file simply isn't being loaded, or that it doesn't work with JSX's embedded XML?  You should be able to test the former by adding an :echo or an :echom somewhere in the ftplugin file.  (I would be surprised if this were the case... if you have an ftplugin/javascript.vim file, it should work for all JS and JSX files.)

If it's just that your plugin itself doesn't support JSX, I'm not sure how I can help you short of hacking on your plugin.

**Comment by selimober** [comment]
@mxw — Thanks for helping. The plugin I mentioned is in fact a JSX plugin. So it does support JSX.

1- I did put an `:echo "xmledt running...` on `ftplugin/javascript.vim` file as you instructed. If I disable the jsx plugin, thus the filetype stays as javascript, I see the output of `:echo`. But not when the jsx plugin enabled, i.e. when the file type is `javascript.jsx`

2- Now, even with the jsx plugin disabled, and "xmledit running..." message displayed, tags are not automatically closed in javascript files, I mean when I close `<p` with `>`, it doesn't turn into a `<p></p>` just like editing an html file with xmledit.

Thanks for trying to help anyways..

**Comment by mxw** [comment]
That behavior is inconsistent with how vim is supposed to behave... when you set a filetype to be foo.bar.baz, then ftplugin/foo.vim, ftplugin/bar.vim, and ftplugin/baz.vim should all get run, and this is the behavior I see...

**Comment by sukima** [comment]
![say wha?](http://www.quickmeme.com/img/c7/c77d2c3aa739d15cd12d672843bd30c22db2fdcd11141bbca03a34f5c16999ef.jpg)

You can bundle filetypes with a dot?! I had no idea you could to that. I thought you needed one filetype and that's it. Hence why this repo suggests creating a symlink to each filetype you want it enable for.

**Comment by mxw** [comment]
Yeah—Unless this package does some weird and narrow filetype-matching, in principle, you can incorporate the syntax, indent, and ftplugin files for any number of filetypes by just setting an aggregate filetype.  (I don't see this documented anywhere in vim's help files, but it definitely works, and there are a few StackOverflow questions about it.)
